src/dmd/expressionsem.d

```diff
@@ -10903,9 +10903,9 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
              */
             static auto extractTypeTupAndExpTup(Expression e)
             {
-                static struct Result { bool ttEmpty; TupleExp te; }
+                static struct Result { bool ttEmpty; bool te; }
                 auto tt = e.op == TOK.type ? e.isTypeExp().type.isTypeTuple() : null;
-                return Result(tt && ((tt.arguments && !tt.arguments.dim) || !tt.arguments), e.isTupleExp());
+                return Result(tt && ((tt.arguments && !tt.arguments.dim) || !tt.arguments), e.isTupleExp() !is null);
```

Comment:
It would be more concise to have this as
```
tt && (!tt.arguments || !tt.arguments.dim)
```

Reply:
indeed. 

Request for auto-merge label, @thewilsonator, due to force push.